core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -62,4 +65,12 @@ private[history] abstract class ApplicationHistoryProvider {
    */
   def getConfig(): Map[String, String] = Map()
 
+  /**
+   * Writes out the event logs to the output stream provided. The logs will be compressed into a
```

Comment:
Yes, we can pass in a zip stream, but there is no real way to enforce the content formats (since each provider might use a different format). Anyway with legacy files, we'd have multiple files zipped (since for legacy files all attempts are the same files, I guess it is fine to just write each of the files as is. I will make that change to be consistent.